src/background/controller/wallet.ts

```diff
@@ -119,6 +119,53 @@ export class WalletController extends BaseController {
     await contract.approve(spender, amount);
   };
 
+  approveNFT = async ({
+    chainServerId,
+    contractId,
+    spender,
+    abi,
+    tokenId,
+    isApprovedForAll,
+  }: {
+    chainServerId: string;
+    contractId: string;
+    spender: string;
+    abi: 'ERC721' | 'ERC1155';
+    isApprovedForAll: boolean;
+    tokenId: string | null | undefined;
+  }) => {
+    const account = await preferenceService.getCurrentAccount();
+    if (!account) throw new Error('no current account');
+    const chainId = Object.values(CHAINS)
+      .find((chain) => chain.serverId === chainServerId)
+      ?.id.toString();
+    if (!chainId) throw new Error('invalid chain id');
+    buildinProvider.currentProvider.currentAccount = account.address;
+    buildinProvider.currentProvider.currentAccountType = account.type;
+    buildinProvider.currentProvider.currentAccountBrand = account.brandName;
+    buildinProvider.currentProvider.chainId = chainId;
+    const provider = new ethers.providers.Web3Provider(
+      buildinProvider.currentProvider
+    );
+    const signer = provider.getSigner();
+    if (abi === 'ERC721') {
+      const contract = new Contract(contractId, ERC721ABI, signer);
+      if (isApprovedForAll) {
+        await contract.setApprovalForAll(spender, false);
+      } else {
+        await contract.approve(
+          '0x0000000000000000000000000000000000000000',
```

Comment:
If you set amount fixed 0, this function better called `unApproveNFT` or `revokeNFTApprove` others better make amount as a parameter